src/main/java/io/split/android/client/impressions/StoredImpressions.java

```diff
@@ -26,4 +29,22 @@ public String id() {
     public List<TestImpressions> impressions() {
         return impressions;
     }
+
+    public int  getAttempts(){
+        return attempts;
+    }
+
+    public void addAttempt(){
+        attempts++;
+    }
+
+    public boolean isDeprecated() {
+        long diff = System.currentTimeMillis() - timestamp;
+
+        long oneDayMillis = 3600 * 1000;
```

Comment:
this should be a configuration. More likely private so it cannot be changed, but still a configuration

Reply:
Changed as isOutdated